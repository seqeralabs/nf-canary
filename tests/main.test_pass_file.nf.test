nextflow_process {

    name "Test Process TEST_PASS_FILE"
    script "main.nf"
    process "TEST_PASS_FILE"

    test("TEST_PASS_FILE") {

        when {
            process {
                """
                input[0] = Channel.of("alpha").collectFile(name: 'sample.txt', newLine: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
