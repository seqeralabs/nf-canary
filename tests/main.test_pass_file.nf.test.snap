{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "out.txt:md5,9f9f90dbe3e5ee1218c86b8839db1995"
                ],
                "outfile": [
                    "out.txt:md5,9f9f90dbe3e5ee1218c86b8839db1995"
                ]
            }
        ],
        "timestamp": "2023-10-04T11:56:24.966198"
    }
}